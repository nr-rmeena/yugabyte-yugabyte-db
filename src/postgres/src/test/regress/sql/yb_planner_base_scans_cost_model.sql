-- This file tests the YB CBO base scans cost model in non-colocated tables.
SET yb_enable_base_scans_cost_model = ON;
SET yb_enable_optimizer_statistics = ON;
CREATE TABLE test (v1 INT, v2 INT, v3 INT);
CREATE INDEX test_index ON test ((v1) HASH, v2 ASC) INCLUDE (v3);
EXPLAIN (COSTS OFF) SELECT * FROM test WHERE v2 > 100;
SELECT * FROM test WHERE v2 > 100;
DROP TABLE test;
