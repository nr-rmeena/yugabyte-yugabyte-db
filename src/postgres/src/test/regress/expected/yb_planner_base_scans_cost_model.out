-- This file tests the YB CBO base scans cost model in non-colocated tables.
SET yb_enable_base_scans_cost_model = ON;
SET yb_enable_optimizer_statistics = ON;
CREATE TABLE test (v1 INT, v2 INT, v3 INT);
CREATE INDEX test_index ON test ((v1) HASH, v2 ASC) INCLUDE (v3);
EXPLAIN (COSTS OFF) SELECT * FROM test WHERE v2 > 100;
                QUERY PLAN
------------------------------------------
 Index Only Scan using test_index on test
   Index Cond: (v2 > 100)
(2 rows)

SELECT * FROM test WHERE v2 > 100;
 v1 | v2 | v3
----+----+----
(0 rows)

DROP TABLE test;
